### ISSUE TITLE
Sort entries in project.yaml

### ISSUE THREAD

**Issue by aNickzz** [craftcms/cms#5147]
### Description

Semi-frequently, making changes in the CP causes trivial changes to the `project.yaml`. I propose that the `project.yaml` properties are kept in alphabetical order to prevent this.

`fieldGroup` and `contentColumnType` have switched places:

```diff
@@ -134,8 +134,8 @@ fields:
       maxBlocks: ''
       contentTable: '{{%matrixcontent_sociallink}}'
       propagationMethod: all
-    fieldGroup: 24ea6a12-cfe8-4111-82a5-3c23dde3dbdd
     contentColumnType: string
+    fieldGroup: 24ea6a12-cfe8-4111-82a5-3c23dde3dbdd
   0fed7a20-a1d7-4f8e-b8ed-7b5ca4a988da:
     name: Person
     handle: keyContacts
```

`fieldGroup` and `contentColumnType` have switched places here too:

```diff
@@ -1430,8 +1417,14 @@ matrixBlockTypes:
               value: linkedin
               imageUrl: 'https://www.linkedin.com/favicon.ico'
               default: ''
-        fieldGroup: null
+            -
+              label: Instagram
+              showLabel: '1'
+              value: instagram
+              imageUrl: 'https://www.instagram.com/favicon.ico'
+              default: ''
         contentColumnType: text
+        fieldGroup: null
       7c43b838-e5b2-41d9-b652-f4c7ffee9aac:
         name: 'Link URL'
         handle: linkUrl
```

It would be better to order the properties alphabetically so the changes in the project config git history reflect the changes in the project config.

### Steps to reproduce

In this case, adding a new option to a field (a `supercool\buttonbox\fields\Buttons` field) made one siginifiant change and 2 trivial changes to the yaml.

### Additional info

- Craft version: Craft Pro 3.3.11
- PHP version: 7.0.33
- Database driver & version: MySQL 5.7.27
- Plugins & versions: Available on request

**Comment by brandonkelly** [comment]
This is how we did it originally, but there are some settings that are relying on the order of associative arrays (like the `columns` setting on Table fields), which break if the Project Config automatically sorts all keys. But I think we should be able to come up with a more nuanced solution for Craft 3.4.

**Comment by aNickzz** [comment]
yeah lists would have to be kept in order, but dictionaries aren't ordered objects (which is why the order changes when writing to the yaml. If there are associative arrays that need to preserve order it should/could be stored as a list of key-value pairs

**Comment by andris-sevcenko** [comment]
Alright, just pushed this for merging into 3.4.

**Comment by brandonkelly** [comment]
This is resolved for 3.4 via 8bbe3b29d3bcb2ac8129c24f358c9a062465d22c.
